dev/tests/integration/testsuite/Magento/Catalog/Model/Product/Option/Type/TextTest.php

```diff
@@ -0,0 +1,104 @@
+<?php
+/**
+ * Copyright © Magento, Inc. All rights reserved.
+ * See COPYING.txt for license details.
+ */
+
+namespace Magento\Catalog\Model\Product\Option\Type;
+
+use Magento\Catalog\Model\Product\Option;
+
+/**
+ * Test for customizable product option with "Text" type
+ */
+class TextTest extends \PHPUnit\Framework\TestCase
+{
+    /**
+     * @var Text
+     */
+    protected $model;
+
+    /**
+     * @var \Magento\Framework\ObjectManagerInterface
+     */
+    private $objectManager;
+
+    /**
+     * {@inheritDoc}
+     */
+    protected function setUp()
+    {
+        $this->objectManager = \Magento\TestFramework\Helper\Bootstrap::getObjectManager();
+        $this->model = $this->objectManager->create(
+            Text::class
+        );
+    }
+
+    /**
+     * Check if newline symbols are normalized in option value
+     *
+     * @dataProvider optionValueDataProvider
+     * @param array $productOptionData
+     * @param string $optionValue
+     * @param string $expectedOptionValue
+     */
+    public function testNormalizeNewlineSymbols(
+        array $productOptionData,
+        string $optionValue,
+        string $expectedOptionValue
+    ) {
+        $productOption = $this->objectManager->create(
+            Option::class,
+            ['data' => $productOptionData]
+        );
+
+        $this->model->setOption($productOption);
+        $this->model->setUserValue($optionValue);
+        $this->model->validateUserValue([]);
+
+        $this->assertSame($expectedOptionValue, $this->model->getUserValue());
+    }
+
+    /**
+     * Data provider for testNormalizeNewlineSymbols
+     *
+     * @return array
+     */
+    public function optionValueDataProvider()
+    {
+        return [
+            [
+                // $productOptionData
```

Comment:
Instead of adding comments, just replace that with:
```php
[self::STUB_OPTION_DATA, "string \r\n string", "string \n string"],
```
And extract `['id' => 11, 'type' => 'area']` to private const.